app/models/energy/ScheduleRuleset.py

```diff
@@ -0,0 +1,207 @@
+    precipitation_rate: 'PrecipitationRate'
+    angle: 'Angle'
+    convection_coefficient: 'ConvectionCoefficient'
+    activity_level: 'ActivityLevel'
+    velocity: 'Velocity'
+    capacity: 'Capacity'
+    power: 'Power'
+    availability: 'Availability'
+    percent: 'Percent'
+    control: 'Control'
+    mode: 'Mode'
+
+
+class ScheduleTypeLimits (BaseModel):
+    """Specifies the data types and limits for values contained in schedules."""
+
+    type: Enum('ScheduleTypeLimits', {'type': 'ScheduleTypeLimits'})
+
+    name: str = Schema(
+        ...,
+        regex=r'^[\s.A-Za-z0-9_-]*$',
+    )
+
+    lower_limit_value: float = Schema(
+        0,
+        description='Lower limit for the schedule type is entered.'
+    )
+
+    upper_limit_value: float = Schema(
+        1,
+        description='Upper limit for the schedule type is entered.'
+    )
+
+    numeric_type: NumericType
+
+    unit_type: UnitType
+
+
+class YesOrNo (str, Enum):
+    no = 'No'
+    yes = 'Yes'
+
+
+class ScheduleDay(BaseModel):
+    """Used to describe the daily schedule for a single simulation day."""
+    type: Enum('ScheduleDay', {'type': 'ScheduleDay'})
+
+    name: str = Schema(
+        ...,
+        regex=r'^[\s.A-Za-z0-9_-]*$',
+    )
+
+    interpolate_to_timestep: YesOrNo = YesOrNo.no
+
+    hour: int = Schema(
+        ...,
+        ge=0,
+        le=23,
+    )
+
+    minute: int = Schema(
+        ...,
+        ge=0,
+        le=59
+    )
+
+    value_until_time: float = Schema(
+        ...
+    )
+
+
+class ScheduleRuleset (BaseModel):
+    """Used to define a schedule for a default day, further described by ScheduleRule."""
+
+    type: Enum('ScheduleRuleset', {'type': 'ScheduleRuleset'})
+
+    name: str = Schema(
+        ...,
+        regex=r'^[\s.A-Za-z0-9_-]*$',
+    )
+
+    schedule_type_limits: ScheduleTypeLimits
+
+    default_day_schedule: List[ScheduleDay]
+
+    summer_designday_schedule: List[ScheduleDay]
+
+    winter_designday_schedule: List[ScheduleDay]
+
+
+class ScheduleRule(BaseModel):
+    """A set of rules assigned to schedule ruleset for specific periods of time and for
+  particular days of the week according to a priority sequence."""
+
+    type: Enum('ScheduleDay', {'type': 'ScheduleDay'})
+
+    name: str = Schema(
+        ...,
+        regex=r'^[\s.A-Za-z0-9_-]*$'
+    )
+
+    schedule_rule_set: List[ScheduleRuleset]
+
+    rule_order: int = Schema(
+        ...,
+        ge=0
+    )
+
+    schedule_day: List[ScheduleDay]
+
+    apply_sunday: YesOrNo = YesOrNo.no
+
+    apply_monday: YesOrNo = YesOrNo.no
+
+    apply_tuesday: YesOrNo = YesOrNo.no
+
+    apply_wednesday: YesOrNo = YesOrNo.no
+
+    apply_thursday: YesOrNo = YesOrNo.no
+
+    apply_friday: YesOrNo = YesOrNo.no
+
+    apply_saturday: YesOrNo = YesOrNo.no
+
+    apply_holiday: YesOrNo = YesOrNo.no
+
+    date_specification_type: DateSpecificationType = DateSpecificationType.date_range
+
+    start_month: int = Schema(
```

Comment:
I think it would be better to use a real date type here.  @chriswmackey @mostaphaRoudsari how do you feel about requiring all dates be fully qualified dates with years?

Reply:
That seems like a good idea but we have to think about how it should be implemented.  If we have date objects, they will likely get used in a lot of different places and it's better to have just one rather than separate copies for E+ schedules, Annual daylight schedules, etc.  This is why we have date objects within ladybug core that get used by a lot of different parts of the insects.  Let's discuss this more in the meeting today.